packaging/python/build_wheels.bash

```diff
@@ -118,17 +118,17 @@ build_wheel_osx() {
     echo " - Installing build requirements"
     cp packaging/python/build_requirements.txt my_requirements.txt
 
+    CMAKE_DEFS="NRN_MPI_DYNAMIC=$3"
+    if [ "$USE_STATIC_READLINE" == "1" ]; then
+      CMAKE_DEFS="$CMAKE_DEFS,NRN_BINARY_DIST_BUILD=ON,NRN_WHEEL_STATIC_READLINE=ON"
+    fi
+
```

Comment:
The rearrangement changes the value of `CMAKE_DEFS` when `$2 == "coreneuron"`. Why versions correct; and if they're not why didn't we notice?

Reply:
I consider the previous version a bug - why set `CMAKE_DEFS` and then unconditionally overwrite it?

`LINK_AGAINST_PYTHON` affects `nmodl` compilation and linkage. I'm not sure it would be notable in the majority of the cases.

Reply:
The reason this is interesting is because, accidentally, we're not setting `LINK_AGAINST_PYTHON` in the one usecase where we need it. Because this flag causes a lot of pain, it would be very good news, if we mistakenly added this complexity.